From: Tommy Webb <tommy@calyxinstitute.org>
Date: Tue, 29 Aug 2023 10:10:47 -0400
Subject: Override Cromite default site settings states

Autoplay: WebView Default ON, Browser still default off
WebGL: Default ON

Change-Id: Ia465ef97d29c8a43f8da7be7d919ba6aad8f95bf
---
 .../core/browser/bromite_content_settings/webgl.inc           | 2 +-
 .../renderer/core/execution_context/execution_context.cc      | 4 ++--
 third_party/blink/renderer/core/html/media/autoplay_policy.cc | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/components/content_settings/core/browser/bromite_content_settings/webgl.inc b/components/content_settings/core/browser/bromite_content_settings/webgl.inc
--- a/components/content_settings/core/browser/bromite_content_settings/webgl.inc
+++ b/components/content_settings/core/browser/bromite_content_settings/webgl.inc
@@ -1,4 +1,4 @@
-  Register(ContentSettingsType::WEBGL, "webgl", CONTENT_SETTING_BLOCK,
+  Register(ContentSettingsType::WEBGL, "webgl", CONTENT_SETTING_ALLOW,
            WebsiteSettingsInfo::SYNCABLE,
            /*allowlisted_schemes=*/{},
            /*valid_settings=*/{CONTENT_SETTING_ALLOW,
diff --git a/third_party/blink/renderer/core/execution_context/execution_context.cc b/third_party/blink/renderer/core/execution_context/execution_context.cc
--- a/third_party/blink/renderer/core/execution_context/execution_context.cc
+++ b/third_party/blink/renderer/core/execution_context/execution_context.cc
@@ -87,8 +87,8 @@ blink::WebContentSettingsClient* GetContentSettingsClientFor(
 bool AllowWebgl(ExecutionContext* context) {
   blink::WebContentSettingsClient* settings = GetContentSettingsClientFor(context);
   if (settings)
-    return settings->AllowContentSetting(ContentSettingsType::WEBGL, false);
-  return false;
+    return settings->AllowContentSetting(ContentSettingsType::WEBGL, /*default_value=*/true);
+  return true;
 }
 
 ExecutionContext::ExecutionContext(v8::Isolate* isolate,
diff --git a/third_party/blink/renderer/core/html/media/autoplay_policy.cc b/third_party/blink/renderer/core/html/media/autoplay_policy.cc
--- a/third_party/blink/renderer/core/html/media/autoplay_policy.cc
+++ b/third_party/blink/renderer/core/html/media/autoplay_policy.cc
@@ -417,7 +417,7 @@ bool AutoplayPolicy::IsAutoplayAllowedPerSettings() const {
     return false;
   if (auto* settings_client = frame->GetContentSettingsClient()) {
     return settings_client->AllowContentSetting(
-        ContentSettingsType::AUTOPLAY, /*default_value*/ false);
+        ContentSettingsType::AUTOPLAY, /*default_value*/ true);
   }
   return true;
 }
-- 

